test/test.js

```diff
@@ -0,0 +1,95 @@
+function test(name, fn) {
+	return require('tape')(name, { timeout: 1000 }, fn)
+}
+const mannish = require('../')
+
+test('Basic functionality: should call the right function and get the response', t => {
+	t.plan(2)
+
+	const mediator = mannish()
+
+	mediator.provide('book', function(book) {
+		t.equal(book, 'enchiridion')
+		return Promise.resolve('mathematical')
+	})
+
+	mediator.provide('unrelated book', () => t.fail(`This function shouldn't be called`))
+
+	mediator.call('book', 'enchiridion').then(response => {
+		t.equal(response, 'mathematical')
+		t.end()
+	})
+})
+
+test('Should receive a non-promise response, too', t => {
+	const mediator = mannish()
+
+	mediator.provide('book', () => 'mathematical')
+
+	mediator.call('book', 'enchiridion').then(response => {
+		t.equal(response, 'mathematical')
+		t.end()
+	})
+})
+
+test('Should receive rejected promise', t => {
+	const mediator = mannish()
+
+	mediator.provide('loot', function(location) {
+		t.equal(location, 'from the dungeon please')
+		return Promise.reject(new Error('skeleton'))
+	})
+
+	mediator.call('loot', 'from the dungeon please').catch(err => {
+		t.ok(err instanceof Error, 'err is an Error')
+		t.equal(err.message, 'skeleton')
+		t.end()
+	})
+})
+
+test(`Errors thrown in the provider should result in rejected responses`, t => {
+	const mediator = mannish()
+
+	mediator.provide('loot', function(location) {
+		throw new Error('monster')
+	})
+
+	mediator.call('loot', 'from the dungeon please').catch(err => {
+		t.ok(err instanceof Error, 'err is an Error')
+		t.equal(err.message, 'monster')
+		t.end()
+	})
+})
+
+test('Should work with multiple arguments', t => {
+	const mediator = mannish()
+
+	mediator.provide('taco', function(first, second) {
```

Comment:
A good chunk of the tests were left over from when the module was first written, which is why the string quotes and function definitions aren't consistent everywhere.  I'll clean that up a bit.